source/io/module_eggShell.js

```diff
@@ -70,6 +70,15 @@
         publicAPI.eggShell = {};
 
         // Private Instance Variables (per vireo instance)
+
+        Module.eggShell.readJavaScriptRefNum = function (valueRef) {
```

Comment:
This should also be exposed on publicAPI.eggShell.readJavaScriptRefNum and publicAPI.eggShell.writeJavaScriptRefNum so the ASW side can update those kind of valueRefs

Reply:
Good idea. Will do.